### ISSUE TITLE
Openverse logo link is broken on search pages

### ISSUE THREAD

**Issue by krysal** [WordPress/openverse-frontend#817]
## Description

Currently, clicking on the logo from a search results page is changing the URL but the error page is shown instead of home, without functioning links, the search bar does not work so there is no way to change the page. 

Seems to be related to the `VTeleport` / `VTeleportTarget` components given the console output.

## Reproduction

1. Go to the search results page and click on the logo at the top
2. See that the page is malfunctioning

## Expectation

The logo should redirect to the homepage correctly

## Additional context

Bug introduced in #771.

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

**Comment by obulat** [comment]
The way we handle Teleport destruction is not working properly. Here is the error output for context:
```
Error: VTeleportTarget: skip-to-content beforeDestroy but still has children mounted
    beforeDestroy VTeleportTarget.vue:25
    VueJS 20
[client.js:103:14](webpack:///.nuxt/client.js)
Error: VTeleport: non-existent target skip-to-content on unmount
    beforeDestroy VTeleport.vue:17
    VueJS 38
```

**Comment by krysal** [comment]
Solved in #852.
